### ISSUE TITLE
Intermittent test failure: sqlite3.OperationalError: database table is locked: _enrichment_jobs

### ISSUE THREAD

**Issue by simonw** [datasette/datasette-enrichments-re2#4]
e.g. in https://github.com/datasette/datasette-enrichments-re2/actions/runs/7037458342/job/19152289050

```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/datasette_enrichments/__init__.py", line 219, in run_enrichment
    await db.execute_write(
  File "/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/datasette/database.py", line 113, in execute_write
    results = await self.execute_write_fn(_inner, block=block)
  File "/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/datasette/database.py", line 167, in execute_write_fn
    raise result
  File "/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/datasette/database.py", line 189, in _execute_writes
    result = task.fn(conn)
  File "/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/datasette/database.py", line 110, in _inner
    return conn.execute(sql, params or [])
sqlite3.OperationalError: database table is locked: _enrichment_jobs
=========================== short test summary info ============================
FAILED tests/test_enrichments_re2.py::test_re2[post2-expected2] - asyncio.exceptions.TimeoutError
========================= 1 failed, 4 passed in 2.42s ==========================
```

**Comment by simonw** [comment]
That's this code here: https://github.com/datasette/datasette-enrichments/blob/e3b220d52b1a896ea75dba53f9a409aeed511471/datasette_enrichments/__init__.py#L216-L227

```python
        async def run_enrichment():
            next_cursor = job["next_cursor"]
            # Set state to running
            await db.execute_write(
                """
                update _enrichment_jobs
                set status = 'running'
                where id = ?
                """,
                (job["id"],),
            )
```

**Comment by simonw** [comment]
This is weird, because the `db.execute_write()` mechanism sends writes through an in-memory queue to avoid any chance of them running at the same time in a way that would suffer from locked tables.

**Comment by simonw** [comment]
I'm not happy about this fix. I switched from using an in-memory named database:

https://github.com/datasette/datasette-enrichments-re2/blob/4e425f8e5c8c26f65d84b98b93f5058c7fbec011/tests/test_enrichments_re2.py#L7-L17

To an on-disk database:

https://github.com/datasette/datasette-enrichments-re2/blob/2d1e767b6024d78a558cb88c6e39af135d1f366a/tests/test_enrichments_re2.py#L8-L16

I did that because I had a hunch that switching to WAL mode might help the problem, and then I spotted on the SQLite forum that in-memory databases cannot use WAL mode: https://sqlite.org/forum/info/a44bddd65750d084

It does seem to have fixed the intermittent test failure, but I'm really unhappy because I want to be able to reliably use in-memory databases without running into this problem.

I'm also not at all sure that my guess as to what was wrong here is correct.
